raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -931,35 +930,53 @@ def handle_init(
     return TransitionResult(iteration.new_state, events)
 
 
-def handle_block(channelidentifiers_to_channels, state, state_change, block_number):
+def handle_block(
+        mediator_state: MediatorTransferState,
+        state_change: Block,
+        channelidentifiers_to_channels: typing.ChannelMap,
+        pseudo_random_generator: random.Random,
+        block_number: typing.BlockNumber,
+):
     """ After Raiden learns about a new block this function must be called to
     handle expiration of the hash time locks.
     Args:
         state (MediatorTransferState): The current state.
     Return:
         TransitionResult: The resulting iteration
     """
-    assert state_change.block_number == block_number
+    balance_proof = mediator_state.transfers_pair[0].payee_transfer.balance_proof
```

Comment:
a mediator may have multiple transfer pairs, all of them need to be cleaned.

change this to a loop in the `mediator_state.transfers_pair`